### PR TITLE
Versioning feature added for badges

### PR DESCRIPTION
**Pull request by nickinack**
Added Papertrail Versioning for AIcrowd badges!

### CODE REVIEW

File: app/admin/paper_trail/version.rb [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
   show do
     columns do
-      link_to 'Back to Submission', admin_challenge_submission_path(paper_trail_version.item.challenge, paper_trail_version.item)
+      if paper_trail_version.item_type == "Submission"
```
Comment (piotrekpasciak) [inline app/admin/paper_trail/version.rb:6]: You can implement the `Back to` link like that:
`link_to "Back to #{paper_trail_version.item_type}", request.referer`

File: app/admin/aicrowd_badges.rb [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+  sidebar "Badge versions", only: [:show, :edit] do
+    ul do
+      aicrowd_badge.versions.reverse.each do |version|
+        li link_to "#{version.created_at}", admin_paper_trail_version_path(version)
+      end
+    end
+  end
+
+  controller do
```
Comment (piotrekpasciak) [inline app/admin/aicrowd_badges.rb:12]: Are you sure you need this block?  It seems to not be needed after you've used my implementation of PaperTrail versions.

File: app/models/aicrowd_user_badge.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class AicrowdUserBadge < ApplicationRecord
+  has_paper_trail
```
Comment (piotrekpasciak) [inline app/models/aicrowd_user_badge.rb:2]: You didn't implement admin page for these.  Is it needed?

**Comment by nikhilrayaprolu** [comment]
Hey can you add some screenshots!

**Pull request merged**